### ISSUE TITLE
How to build with CMake on macOS?

### ISSUE THREAD

**Issue by tiendq** [sqlitebrowser/sqlitebrowser#1555]
I just tried to build this project on macOS for the first time, I found there are different ways to build for different OSes e.g I saw both `cmake`, `qmake`, and `make`.

I tried make a `build` folder and use `CMake` (on macOS) but it didn't work (it didn't find `Qt5.make`), while it's instructed way for `Ubuntu Linux`. How can I use `CMake` on macOS? or better for building on all supported OSes?

Thanks,

**Comment by tiendq** [comment]
@justinclift I did, if not how I know it uses `qmake`.

**Comment by tiendq** [comment]
I got same build issue at raised in #1439, `QScintilla` is rebuilt even it existed (via `brew install qscintilla2`). Build errors are naming issues between `qsciscintillabase.h` and `Scintilla.h`, for example:

```
/GitHub/sqlitebrowser/libs/qscintilla/Qt4Qt5/Qsci/qsciscintillabase.h:70:9: error: expected identifier
        SCI_START = 2000,
        ^
/GitHub/sqlitebrowser/libs/qscintilla/Qt4Qt5/../include/Scintilla.h:48:19: note: expanded from macro 'SCI_START'
#define SCI_START 2000
```

My system is macOS 10.12.6 and clang-900.0.39.2

**Comment by justinclift** [comment]
Ouch.  This isn't something I can help with personally, so hopefully one of our other Community members has the time/interest to investigate. :smile:

**Comment by justinclift** [comment]
@tiendq With that "Qt path need to be supplied in CMake command ...", is the example `cmake -DCMAKE ...` line definitely correct?

Asking because it has `cmake` inside what looks like a `Qt` installation.  Should it be `qmake` (the Qt alternative to CMake) instead? :smile:

**Comment by tiendq** [comment]
@justinclift Yes, I copied it from terminal :). Qt was installed by `brew` the path actually has many folders inside, they contains `.cmake` files for each Qt components.

**Comment by justinclift** [comment]
No worries. :smile:

We have a (very old, slow) mac mini used for as our nightly build server.  In theory I should be able to investigate this in depth using that, then update our docs as appropriate.

It won't be this week though, as my schedule has become rather full very quickly.  And it's just the start of the week! :confounded:

**Comment by MKleusberg** [comment]
@tiendq I'll leave this to @justinclift since I don't have access to a Mac to try this out myself. As for the qscintilla question: it actually should prefer the system installation of qscintilla over the version we include in our repository - so in theory it should just work. If it doesn't that means we may need to adjust the ```HINTS``` lines in the ```cmake/FindQScintilla.cmake``` file to include the path brew uses.

**Comment by tiendq** [comment]
@MKleusberg Providing path information (as I did) is actually the way it needs to work with Qt on macOS ([see this](https://blog.kitware.com/cmake-finding-qt5-the-right-way/)). I would make a PR for this small change in docs. Regarding `qscintilla`, if it's what you prefer I'd find to make it work with system installation too.

**Comment by MKleusberg** [comment]
Thanks, @tiendq! A PR for that would be awesome :smile: With the qscintilla library, any way to make it work with the system installation is welcomed, be it automatically or via a command line parameter.
